src/taipy/core/config/data_node_config.py

```diff
@@ -267,15 +296,26 @@ def _update(self, as_dict, default_section=None):
         self._storage_type = as_dict.pop(self._STORAGE_TYPE_KEY, self._storage_type)
         if self._storage_type is None and default_section:
             self._storage_type = default_section.storage_type
+
         self._scope = as_dict.pop(self._SCOPE_KEY, self._scope)
         if self._scope is None and default_section:
-            self._scope = default_section.scope
+            if default_section.scope:
```

Comment:
For now, the Scope is always assigned as the `default_section.scope` if defined (even with different storage_type).

If we only want the Scope to be `default_section.scope` when the storage_type is the same, add ` and self._storage_type == default_section.storage_type:` condition here.